### ISSUE TITLE
A bunch of custom emojis that I've made this summer

### ISSUE THREAD

**Issue by xavizardKnight** [hfg-gmuend/openmoji#475]
Heyas! This summer and these lasts weeks, I have been working on quite a few icons and emojis in the style of OpenMoji, as a way to practice SVGs with Inkscape (as I mentioned in the issue #462 I opened before).

I've made these emojis for my projects, and I've shared them in [this GitLab repo](https://gitlab.com/XavizardKnight/openmoji-xk-extensions). But because some of these emojis have also been requested here in different issues, or I feel might be nice to have here, I want to suggest moving some of them here, so more people can use them.

Here are the emojis that, I think, might be cool to move over to here:

![a](https://github.com/hfg-gmuend/openmoji/assets/35200790/08e57bf3-9375-4e1b-858f-a80e8101ebfd)

- Flags of the Valencian Community, Balearic Islands, Aragon and Murcia are subdivisions of Spain, and have Unicode sequences. They can be placed in the extras-unicode/subdivision-flag category. These flags were requested by #416 , although there are more Spain subdivision flags left to do (I may create them in the near future).
- The Climate Change flag, Disability flag, Disability Pride flag and the Vegan flag are initiative flags that I think that might fit your non-unicode flags about social and critical issues. All four are public domain and/or CC0.
- Motorbike helmet. I've always wondered why the standard emoji sets have different types of helmets, but not one for motorbikes. The current design in colour has a black bolt as the pivot point of the visor, the black/white versions have this black bolt removed, as I think it looks better without it when the colour layer is hidden.
![2023-11-14_20-32](https://github.com/hfg-gmuend/openmoji/assets/35200790/35235585-ef76-4766-927b-beab6ff286ba)

- The Guy Fawkes Mask, also known as the Anonymous Mask, is a white mask (although light grey in this emoji) with black details. The original mask has a light red shadow on the cheeks, but I left it out in this emoji because it looked too crowded. The current design also has the lines above the nose as grey shadows, so they are only visible in the coloured output and not on the white/black output, maybe I should change that…
![2023-11-14_20-33](https://github.com/hfg-gmuend/openmoji/assets/35200790/0cfffa70-f896-4569-b654-4c5ece2f28c2)

- Chrome Canary and Firefox Nightly/Developer are recoloured versions of the Chrome/Chromium and Firefox emojis representing different prerelease versions of said browsers. Please don't hesitate to reject these icons if you feel that they are too niche for this repo.
- The icons of the C, C++, C#, Markdown, JavaScript, TypeScript and WebAssembly, for satisfying issue #468 that requested some programming language icons. More on that in a note below.
- SVG icon. Because of the form it has, I've made that the white/black versions use line-supplement to draw a different and more minimalist SVG icon; although this renders the standard line layer of this SVG essentially useless. And now that I think about it, maybe I should make this emoji smaller because I think it's too big.
![2023-11-14_20-34](https://github.com/hfg-gmuend/openmoji/assets/35200790/e6b1b4b6-4cb8-4b58-822c-85d4793b6ac4)

- The WinRAR icon is using the books emoji with the WinRAR colours and a brown strap around them; I tried some designs for the belt buckle like the original logo has, but they didn't convince me, and I'm also not much of a fan with the current “empty” version, if someone has an idea to improve it… The WinRAR webpage has this section (https://www.rarlab.com/logo.htm) where fan-made logo submissions are showcased. If we add this emoji to this repo collection, we can email Rarlab, so they can display it on their webpage. 

A couple of notes:
1) As referred in the C/C++/C#/SVG/Markdown/JS/TS/WASM bullet point, the programming icons that I suggest here are all public domain. Other interesting programming icons to add might be the Ruby icons or the classic HTML5/CSS3/JS shields, but those require attribution. How would you add attribution for emojis/icons and brands that might require them? Currently, your library is empty of descriptions for everything non-unicode.
2) I've been having some problems with the Visual Dark Mode Check of the OpenMoji tester with some of my emojis, where the white outline will in some emojis be placed very weirdly for no reason. May I ask how this outline works? I have currently 3 emojis whose white outline misbehaves for reasons I cannot understand (one is the motorbike helmet, the other two are in my repo).

I'll be willing to help to move any or all of these emojis to here (or any other from my repo you feel might also fit better in this repo).

Thank you and have a nice day!

**Comment by xavizardKnight** [comment]
I wanted to add direct links to the source svg files per emoji I mentioned above, but I forgot; and the message is so big that the edit UI doesn't seem to load, so I'll add the links here.

Just a disclaimer, sometimes GitLab's SVG viewer does weird things, and might now show the SVG properly; please download it and view it with your local viewer or editor of choice to avoid surprises.

<details>
<summary><b>22 links collapsed to avoid flooding</b></summary>

- Flag of Valencian Community: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/pa%C3%AFsos-catalans/pa%C3%ADs-valenci%C3%A0.svg?ref_type=heads
- Flag of Balearic Islands: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/pa%C3%AFsos-catalans/illes-balears.svg?ref_type=heads
- Flag of Aragon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/pa%C3%AFsos-catalans/arag%C3%B3.svg?ref_type=heads
- Flag of Murcia: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/pa%C3%AFsos-catalans/m%C3%BArcia.svg?ref_type=heads


- Climate Change Flag: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/initiatives/climate.svg?ref_type=heads
- Disability Flag: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/initiatives/disability.svg?ref_type=heads
- Disability Pride Flag: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/initiatives/disability-pride.svg?ref_type=heads
- Vegan Flag: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/flags/initiatives/vegan.svg?ref_type=heads


- Motorbike Helmet: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/objects/clothing/motorbike-helmet.svg?ref_type=heads
- Guy Fawkes Mask: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/objects/clothing/guy-fawkes-mask.svg?ref_type=heads


- Chrome Canary Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/chrome-canary.svg?ref_type=heads
- Firefox Nightly Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/firefox-nightly.svg?ref_type=heads
- Firefox Developer Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/firefox-developer.svg?ref_type=heads


- C Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/c.svg?ref_type=heads
- C++ Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/cplusplus.svg?ref_type=heads
- C# Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/csharp.svg?ref_type=heads
- SVG Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/svg.svg?ref_type=heads
- Markdown Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/markdown.svg?ref_type=heads
- JavaScript Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/javascript.svg?ref_type=heads
- TypeScript Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/typescript.svg?ref_type=heads
- WebAssembly Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/webassembly.svg?ref_type=heads
- WinRAR Icon: https://gitlab.com/XavizardKnight/openmoji-xk-extensions/-/blob/master/src/brand/winrar.svg?ref_type=heads

</details>

Sorry If these are too many links :sweat_smile:

**Comment by b-g** [comment]
Dear @xavizardKnight, my apologies for the super low reply! It has been very busy here with mid term exams ... and on top of this a cold caught me badly as well. Uff.

Basically this is all pure gold! 😻 We would be more than happy to have nearly all your beautiful contributions!! 

Comments:

* ✅ Flags of the Valencian Community, Balearic Islands, Aragon and Murcia are subdivisions of Spain

* ❌ The Climate Change flag, Disability flag, Disability Pride flag and the Vegan flag are initiative flags.

We changed a while our contribution guidelines and no longer are accepting any flags without an official unicode definition. Sorry! But arguing e.g. to include a flag like Kurdistan or not is too tough. We resignated at some point as we simply couldn't moderate it. Hence we won't be able to include those (even if I may add, personally I would too). I hope you understand.

* ✅ Motorbike helmet

* ✅ The Guy Fawkes Mask

* ✅ Chrome Canary and Firefox Nightly/Developer 

* ✅ The icons of the C, C++, C#, Markdown, JavaScript, TypeScript and WebAssembly
* ✅ SVG icon
* ✅ WinRAR (+ good call to let Rarlab know, to hopefully get listed there.)

> As referred in the C/C++/C#/SVG/Markdown/JS/TS/WASM bullet point, the programming icons that I suggest here are all public domain. Other interesting programming icons to add might be the Ruby icons or the classic HTML5/CSS3/JS shields, but those require attribution. How would you add attribution for emojis/icons and brands that might require them? 

Currently, your library is empty of descriptions for everything non-unicode.
To be honest, not very keen to implement a new attribution mechanism. We have so much on our plate e.g. dark mode, which seems a lot more important and are resources are limited.

> I've been having some problems with the Visual Dark Mode Check of the OpenMoji tester with some of my emojis, where the white outline will in some emojis be placed very weirdly for no reason. May I ask how this outline works? I have currently 3 emojis whose white outline misbehaves for reasons I cannot understand (one is the motorbike helmet, the other two are in my repo).

The idea is to have in the near future an additional step in the build process, which generates automatically the white outlines. The approach is this: https://github.com/hfg-gmuend/openmoji-tester/blob/master/modules/getSvgWithAddedOutline.mjs ... understandable?

> I'll be willing to help to move any or all of these emojis to here (or any other from my repo you feel might also fit better in this repo).

Yay!! Could you prepare a PR with your contributions? I'm planing to release the v15 in December. It would be super cool to include your OpenMojis!! 

All the very best!

**Comment by b-g** [comment]
Hi @xavizardKnight,

> I understand your decision with the initiative flags. I'll start working on adding the rest of the emojis. 

Many thanks for our understanding!

> I've had a busy weekend with family meetups, but I'll try to get the PR done in the upcoming days. It's my first time adding emojis to this repo, so I hope to do it right… 😅

No worries, happy to help or to take at some time if it gets to nitty gritty :)

> I've been looking at the process for adding new emojis; and I assume that the emojis of the motorbike-helmet and the guy-fawkes-mask should be in a clothing subcategory and inside openmoji-extras (in my repo they are in the objects/clothing folders, as I try to do a similar folder structure); but in openmoji-extras there is no clothing subfolder, only an objects subfolder. Should I create a new openmoji-extras/clothing subfolder, or should I place them in the already existing openmoji-extras/objects subfolder?

Please in the `openmoji-extras/objects` subfolder, as two seems not enough to "justify" a new subfolder for clothing. You can run `npm run show-extras-hexcodes`to get the next available hexcode ...

> For what I was able to understand from the code (because my JavaScript skills are near zero; I'm more of a C# guy xD), the outline seems to prefer the color layer rather than the line layer, which is something I was able to (sort of) guess based on how it misbehaved on some of my emojis.

Yes, exactly ...

> Mainly on the motorbike-helment one, which currently looks like this:

This is odd. Happy to look at some point into it and use your motorbike-helment to improve the tool further. However in the meantime it might be the easiest (think quick fix) to simply add a white fill underneath everything, so that the outlining works. Would that be a workaround? (Or add it as it is to the PR – just make sure that all tests are green – and we will take over, I expect a lot of manual tweaking to get the upcoming dark mode really going in the end. One more won't hurt. Uff.)

> As I said, will try to get this PR in the following days. Will let you know if more issues arise.

👍 + Thanks!

**Comment by xavizardKnight** [comment]
Hi!

I've spent this afternoon working on preparing everything. I'll continue tomorrow, and hopefully, I can finish it.

> Please in the openmoji-extras/objects subfolder, as two seems not enough to "justify" a new subfolder for clothing.

Okey, I've already done it this way :+1:

> However in the meantime it might be the easiest (think quick fix) to simply add a white fill underneath everything, so that the outlining works. Would that be a workaround?

I've been playing around with this emoji this afternoon and I managed to come out with something. Notice how in the screenshot that I've sent yesterday, the only part of the outline that works correctly is the bottom dark grey area of the neck of the helmet. So I tried to extend this grey area to the rest of the helmet and it worked!

![2023-12-05_01-36](https://github.com/hfg-gmuend/openmoji/assets/35200790/5c0efc25-295f-4fa6-8ef6-fd987813d667)

I don't understand why the outline registers with this grey area but not with the rest of the shapes in the emoji. This outline can still be improved, as right now, I had to adjust the grey area manually, that's why the outline looks "bumpy" in some areas.

> just make sure that all tests are green

Yep, that's what I'm doing. Before publishing them to my repo, I already make sure all the tests are green, so everything should be already correct; but I'm anyway re-checking everything again just to be safe.
